src/FRP/Event/Time.purs

```diff
@@ -15,3 +17,38 @@ foreign import animationFrame :: Event Unit
 
 -- | Create an event which reports the current time in milliseconds since the epoch.
 foreign import withTime :: forall a. Event a -> Event { value :: a, time :: Number }
+
+-- | Provided an input event and transformation, block the input event for the
+-- | duration of the specified period on each output.
+withBlocking
+  :: forall a b.
+     (Event a -> Event { period :: Number, value :: b })
+  -> Event a
+  -> Event b
+withBlocking process event
```

Comment:
Since this generalizes `debounce` (or whatever we end up calling it), how about naming this something like `debounceWith`?

Reply:
Yep, no worries :D